src/controllers/DynamicFacetRangeQueryController.ts

```diff
@@ -1,33 +1,35 @@
 import { DynamicFacetQueryController } from './DynamicFacetQueryController';
 import { IFacetRequest, IFacetRequestValue } from '../rest/Facet/FacetRequest';
-import { IDynamicFacet } from '../ui/DynamicFacet/IDynamicFacet';
 import { IQuery } from '../rest/Query';
+import { IDynamicFacetRange } from '../ui/DynamicFacet/IDynamicFacetRange';
 
 export class DynamicFacetRangeQueryController extends DynamicFacetQueryController {
-  protected facet: IDynamicFacet;
+  protected facet: IDynamicFacetRange;
 
   public buildFacetRequest(query: IQuery): IFacetRequest {
     return {
       ...this.requestBuilder.buildBaseRequestForQuery(query),
-      preventAutoSelect: this.preventAutoSelection,
       currentValues: this.currentValues,
       numberOfValues: this.numberOfValues,
-      freezeCurrentValues: this.facet.values.hasValues
+      freezeCurrentValues: false,
+      generateAutomaticRanges: !this.areRangesDefined
     };
   }
 
   protected get numberOfValues() {
-    return this.facet.values.hasValues ? this.currentValues.length : this.facet.options.numberOfValues;
+    return this.areRangesDefined ? this.facet.options.ranges.length : this.facet.options.numberOfValues;
```

Comment:
Related to line 15 and this line, what about the use-case where ranges are defined, and automatic ranges is `true`? e.g. sharing a url link with a selected range.

Would the expression `generateAutomaticRanges: !this.areRangesDefined` lead to a Range Facet with only 1 value?
Also, in this scenario, would the number of values still be the configured number of ranges?

Reply:
Maybe areRangesDefined is not clear enough... 
It means ranges are defined in the option of the facet, in that case we use that as the numberOfValues & we never need to get ranges from the index/API.

When you navigate to a URL with a selected range, it will be in the currentValues, but not the range option of the facet